android/app/src/main/java/com/app/edonymyeon/presentation/ui/main/MainActivity.kt

```diff
@@ -30,41 +32,31 @@ class MainActivity : AppCompatActivity() {
         ActivityMainBinding.inflate(layoutInflater)
     }
 
+    private var backClickedTime = 0L
+
+    private val backPressedCallback = object : OnBackPressedCallback(true) {
+        override fun handleOnBackPressed() {
+            if (System.currentTimeMillis() - backClickedTime >= FINISHING_APP_TIME) {
+                backClickedTime = System.currentTimeMillis()
+                binding.root.makeSnackbar(getString(R.string.main_ask_back_click))
+            } else {
+                finish()
+            }
+        }
+    }
+
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         setContentView(binding.root)
+        setOnBackCallback()
         initFragmentContainerView()
         setBottomNavigationView()
+        setAlarmObserver()
         navigateByIntent()
-
-        AlarmService.isAlarmOn.observe(this) {
-            if (it) {
-                binding.bnvMain.menu.findItem(R.id.bottom_menu_alarm)
-                    .setIcon(R.drawable.ic_bottom_nav_alarm_on)
-            }
-        }
-    }
-
-    private fun navigateByIntent() {
-        when (intent.extras?.getString(NOTIFICATION_CLICK_EVENT)) {
-            NOTIFICATION_CLICK_EVENT_POST -> {
-                val postId = (intent.extras?.getString(NOTIFICATION_POST_ID) ?: "0").toLong()
-                val notificationId = (intent.extras?.getString(NOTIFICATION_ID) ?: "0").toLong()
-                navigateToPostDetail(postId, notificationId)
-            }
-
-            NOTIFICATION_CLICK_EVENT_MYPOST -> {
-                navigateToMyPost()
-            }
-        }
     }
 
-    private fun navigateToPostDetail(postId: Long?, notificationId: Long?) {
-        startActivity(PostDetailActivity.newIntent(this, postId ?: 0, notificationId ?: -1))
-    }
-
-    private fun navigateToMyPost() {
-        startActivity(MyPostActivity.newIntent(this))
+    private fun setOnBackCallback() {
+        this.onBackPressedDispatcher.addCallback(this, backPressedCallback)
```

Comment:
```suggestion
    private fun setOnBackCallback() {
        var backClickedTime = 0L
        
        this.onBackPressedDispatcher.addCallback(
            this,
            object : OnBackPressedCallback(true) {
                override fun handleOnBackPressed() {
                    if (System.currentTimeMillis() - backClickedTime >= FINISHING_APP_TIME) {
                        backClickedTime = System.currentTimeMillis()
                        binding.root.makeSnackbar(getString(R.string.main_ask_back_click))
                    } else {
                        finish()
                    }
                }
            },
        )
```

이렇게 멤버변수 수를 줄이는 건 어떨까요?!

Reply:
depth가 너무 깊어지지 않을까 고민했어요 🤔 
다만, 저 callback은 해당 함수에서만 쓰이기 때문에 멤버변수 수를 줄이고 제안해주신 것처럼 표현하면 
코드를 전체적으로 읽을 때 좀 더 가독성 측면에서 좋을 것 같기도 하네요!